### PR TITLE
table: tiny clean up and add logs

### LINKED ISSUE #5179: tables/tables.go: Could it be more easy
I tried to read code about `UpdateRecord`(https://github.com/pingcap/tidb/blob/master/table/tables/tables.go#L235) while the following code make me confused:
```
if col.State != model.StatePublic {
	// If col is in write only or write reorganization state
	// and the value is not default, keep the original value.
	value, err = table.GetColOriginDefaultValue(ctx, col.ToInfo())
	if err != nil {
		return errors.Trace(err)
	}
	cmp, errCmp := oldData[col.Offset].CompareDatum(ctx.GetSessionVars().StmtCtx, &value)
	if errCmp != nil {
		return errors.Trace(errCmp)
	}
	if cmp != 0 {
		value = oldData[col.Offset]
	}
}		
```

It seems the same with
```
if col.State != model.StatePublic {
   value = oldData[col.Offset]
}
```
 Does it ?

### PR DESCRIPTION
**Pull request by zimulala**
Fix #5179 and add logs

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @coocood @winkyao @shenli @tiancaiamao

**Comment by shenli** [comment]
Duplicate with #5185

**Comment by zimulala** [comment]
cc @AndreMouche @hicqu

File: table/tables/tables.go [hunk 1]
```diff
@@ -584,12 +575,14 @@ func (t *Table) removeRowIndices(ctx context.Context, h int64, rec []types.Datum
 		vals, err := v.FetchValues(rec)
 		if vals == nil {
 			// TODO: check this
```
Comment (ngaut) [inline table/tables/tables.go]: Is this a stale TODO ?

Comment (zimulala) [reply table/tables/tables.go]: Nop, If the index is public, I think we need handle it.

**Comment by shenli** [comment]
LGTM

File: table/tables/tables.go [hunk 2]
```diff
@@ -233,19 +233,10 @@ func (t *Table) UpdateRecord(ctx context.Context, h int64, oldData, newData []ty
...
-			if errCmp != nil {
-				return errors.Trace(errCmp)
-			}
-			if cmp != 0 {
-				value = oldData[col.Offset]
-			}
+			// If col is in write only or write reorganization state we should keep the oldData.
+			// Because the oldData must be the orignal data(it's changed by other TiDBs.) or the orignal default value.
+			// TODO: Use newData directly.
+			value = oldData[col.Offset]
```
Comment (AndreMouche) [inline table/tables/tables.go]: Could we  move this update before `rebuildIndices`, although an indexed col would always be public?

File: table/tables/tables.go [hunk 3]
```diff
@@ -584,12 +575,14 @@ func (t *Table) removeRowIndices(ctx context.Context, h int64, rec []types.Datum
 		vals, err := v.FetchValues(rec)
```
Comment (coocood) [inline table/tables/tables.go]: The error is ignored

File: table/tables/tables.go (hunk 2 above)
Comment (zimulala) [reply table/tables/tables.go]: We will remove this code next time, so we should do it now.

**Pull request merged**